osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -151,6 +151,38 @@ public async Task AddItem(MultiplayerPlaylistItem item, MultiplayerRoomUser user
             }
         }
 
+        /// <summary>
+        /// Removes a playlist item from the room's queue.
+        /// </summary>
+        /// <param name="playlistItemId">The item to remove.</param>
+        /// <param name="user">The user removing the item.</param>
+        public async Task RemoveItem(long playlistItemId, MultiplayerRoomUser user)
+        {
+            if (dbFactory == null) throw new InvalidOperationException($"Call {nameof(Initialise)} first.");
+
+            var item = room.Playlist.FirstOrDefault(item => item.ID == playlistItemId);
+
+            if (item == null)
+                throw new InvalidStateException("Item does not exist in the room.");
+
+            if (item == CurrentItem)
+                throw new InvalidStateException("The room's current item cannot be removed.");
```

Comment:
Is there a technical reason for this limitation? Thinking we should allow it is there's nothing stopping us from doing so. If it's only here for the "host only" mode, we should limit it to that.

Reply:
Nothing handles this gracefully right now...

- `updateCurrentItem` assumes there's always one item in the room, expired or otherwise.
- `MultiplayerRoomSettings.PlaylistItemId` is non-nullable.
- "Adding" an item in host-only mode re-uses/updates the current item.
- Client side doesn't gracefully handle this either. Not only the same issues as above but some callbacks like `SettingsChanged` expect to have the `PlaylistItemId` given in the playlist.

I would like to fix that in a future PR, but I just don't feel safe with allowing the current item to be removed with the current state of things.

Reply:
Okay, let's leave it for next time.